mesonbuild/backend/ninjabackend.py

```diff
@@ -2369,7 +2369,8 @@ def get_target_type_link_args(self, target, linker):
             commands += linker.get_pic_args()
             # Add -Wl,-soname arguments on Linux, -install_name on OS X
             commands += linker.get_soname_args(target.prefix, target.name, target.suffix,
-                                               abspath, target.soversion,
+                                               abspath, target.soversion, target.ltversion,
+                                               os.path.join(self.environment.get_prefix(), self.environment.get_libdir()),
```

Comment:
Adding libdir argument here is unnecessary as it is only for the rpath change, as far as I can tell, so should be split into a different commit and PR.

Reply:
IIRC this stuff was necessary for the Gobject introspection bindings to compile properly.

Need to go through this again but a bit short on time...